src/components/BooksList.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { connect, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import actions from '../actions/index';
+import Book from '../containers/Book';
+import CategoryFilter from '../containers/CategoryFilter';
+
+const BooksList = ({ changeFilter }) => {
+  const books = useSelector(state => state.books);
+  const filterState = useSelector(state => state.filter);
+  const handleFilterChange = e => {
+    changeFilter(e.target.value);
+  };
+
+  const filteredBooks = () => {
+    let newBooks;
+    if (filterState === 'All') {
+      newBooks = books;
+    } else {
+      newBooks = books.filter(book => book.category === filterState);
+    }
+    return newBooks;
```

Comment:
Why not return immedatlly in the first if then there won't be a need for the else case. You can return the filtered book immedialty there too. Makes the code more straightforward without unnecessary variables.